### PR TITLE
Fixes #5488 - fixed hook deprecation warning

### PR DESCRIPTION
**Pull request by mbacovsky**
- code from installer executable was moved to hooks
- services details in post-install message printed out on success only

### CODE REVIEW

**Comment by ares** [comment]
s/fixed/fixes/ in commit message (AFAIK our redmine don't match fixed)

File: config/foreman-installer.yaml [hunk 1]
```diff
@@ -7,6 +7,7 @@
 :installer_dir: .
 # Uncomment if you want to load puppet modules from a specific path, $pwd/modules is used by default
 :modules_dir: ./_build/modules
+:hooks_dir: "./hooks"
```
Comment (ares) [inline config/foreman-installer.yaml]: why is this needed? kafo adds `root_dir + '/hooks'` by default

EDIT:
also the option name is hook_dirs and is expected to be an array

**Comment by mbacovsky** [comment]
Commit message fixed, `hooks_dir` removed.

File: hooks/post/10-post_install_message.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# helpers
+def module_enabled?(name)
```
Comment (ares) [inline hooks/post/10-post_install_message.rb]: this is no longer needed, see https://github.com/theforeman/kafo/blob/master/lib/kafo/configuration.rb#L126,  https://github.com/theforeman/kafo/blob/master/lib/kafo/hook_context.rb#L13

you should be able to use `@kafo.config.module_enabled?(name)`

File: hooks/post/10-post_install_message.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  mod = @kafo.module(name)
+  return false if mod.nil?
+  mod.enabled?
+end
+
+def get_param(mod, name)
```
Comment (ares) [inline hooks/post/10-post_install_message.rb]: this is no longer needed, see https://github.com/theforeman/kafo/blob/master/lib/kafo/hook_context.rb#L47

you should be able to use `param('foreman', 'admin_username').value` directly

**Comment by domcleal** [comment]
While I remember, we should check (or just bump to $current... either way) the minimum kafo version and update the packaging to match.

**Comment by mbacovsky** [comment]
Thanks for the hints @ares, helpers were removed.
I'll update the packaging @domcleal, I wouldn't mind if there are volunteers for debs though ;)

**Comment by domcleal** [comment]
Oh sure, that's easy enough.  We have almost up-to-date kafo packages, so it's just bumping the min version.

**Comment by mbacovsky** [comment]
Found out that build task in the Rakefile didn't included hooks directory plus one bug in the exit_code handling. Fixed both, updated

**Comment by mbacovsky** [comment]
Bumped kafo version in rpm package dependences https://github.com/theforeman/foreman-packaging/pull/393

**Comment by ares** [comment]
lgtm I'll have to do the testing

**Comment by ares** [comment]
@mbacovsky would you mind uploading links to scratch builds you used for testing?

**Comment by ares** [comment]
nvm, built myself one and testing with it

**Comment by mbacovsky** [comment]
@ares One is linked in the foreman-packaging PR I mentioned above

**Comment by ares** [comment]
Merged as https://github.com/theforeman/foreman-installer/commit/40fc22eecd3f313dfa63870f9513d54f355b1f86, thanks @mbacovsky!
